packages/elasticsearch-client/src/updateDocument.ts

```diff
@@ -0,0 +1,88 @@
+/**
```

Comment:
If I understand correctly, this is used to PATCH a document, rather than PUT one, in REST terms?

Reply:
Yes that is correct! We could overwrite if desired, but I don't see in which cases that's useful since the only things that can be cleaned are not root properties (hence they wont be filtered), as the "root fields" like timestamps, worker sids etc are always filled.